File: bindings/lua/Polycode/SpriteFrame.lua
class "SpriteFrame"



function SpriteFrame:__getvar(name)
	if name == "coordinates" then
		local retVal = Polycore.SpriteFrame_get_coordinates(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Rectangle"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "anchorPoint" then
		local retVal = Polycore.SpriteFrame_get_anchorPoint(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector2"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "frameID" then
		return Polycore.SpriteFrame_get_frameID(self.__ptr)
	end
end


function SpriteFrame:__setvar(name,value)
	if name == "coordinates" then
		Polycore.SpriteFrame_set_coordinates(self.__ptr, value.__ptr)
		return true
	elseif name == "anchorPoint" then
		Polycore.SpriteFrame_set_anchorPoint(self.__ptr, value.__ptr)
		return true
	elseif name == "frameID" then
		Polycore.SpriteFrame_set_frameID(self.__ptr, value)
		return true
	end
	return false
end


function SpriteFrame:__delete()
	if self then Polycore.delete_SpriteFrame(self.__ptr) end
end

File: bindings/javascript/Polycode/SpriteFrame.js
function SpriteFrame() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.SpriteFrame()
	}
	Object.defineProperties(this, {
		'coordinates': { enumerable: true, configurable: true, get: SpriteFrame.prototype.__get_coordinates, set: SpriteFrame.prototype.__set_coordinates},
		'anchorPoint': { enumerable: true, configurable: true, get: SpriteFrame.prototype.__get_anchorPoint, set: SpriteFrame.prototype.__set_anchorPoint},
		'frameID': { enumerable: true, configurable: true, get: SpriteFrame.prototype.__get_frameID, set: SpriteFrame.prototype.__set_frameID}
	})
}


SpriteFrame.prototype.__get_coordinates = function() {
	var retVal = new Rectangle("__skip_ptr__")
	retVal.__ptr = 	Polycode.SpriteFrame__get_coordinates(this.__ptr)
	return retVal
}

SpriteFrame.prototype.__set_coordinates = function(val) {
	Polycode.SpriteFrame__set_coordinates(this.__ptr, val.__ptr)
}

SpriteFrame.prototype.__get_anchorPoint = function() {
	var retVal = new Vector2("__skip_ptr__")
	retVal.__ptr = 	Polycode.SpriteFrame__get_anchorPoint(this.__ptr)
	return retVal
}

SpriteFrame.prototype.__set_anchorPoint = function(val) {
	Polycode.SpriteFrame__set_anchorPoint(this.__ptr, val.__ptr)
}

SpriteFrame.prototype.__get_frameID = function() {
	return Polycode.SpriteFrame__get_frameID(this.__ptr)
}

SpriteFrame.prototype.__set_frameID = function(val) {
	Polycode.SpriteFrame__set_frameID(this.__ptr, val)
}

Duktape.fin(SpriteFrame.prototype, function (x) {
	if (x === SpriteFrame.prototype) {
		return;
	}
	Polycode.SpriteFrame__delete(x.__ptr)
})
